### PR TITLE
feat: support Dolby Vision profile 8.x (HEVC) and 10.x (AV1) in HLS and DASH 

### PR DESCRIPTION
**Pull request by cosmin**
Support Dolby Vision profile 8.1, 8.2, 8.4, 10.1, 10.4 signaling in HLS and DASH.

adds new option `--use_dovi_supplemental_codecs` (off by default) to use SUPPLEMENTAL-CODECS in HLS and `scte214:supplementalCodecs` and `scte214:supplementalProfiles` for DASH.

To maintain compatibility with existing players the current behavior of using two entries in the manifest remains the default. This will be changed in a future version where `use_dovi_supplemental_codecs` will become on by default.

Adds Dolby Vision compatible brands, 'db1p', 'db2g', 'db4g', 'db4h', 'dby1' based on https://mp4ra.org/#/brands

This is based on and replaces #1255. It was then extended to support DolbyVision profile 10.x signaling for AV1

---

Co-authored-by: Xingzhao Yun <xyun@dolby.com>

### CODE REVIEW

**Comment by cosmin** [comment]
@vish91 I know you expressed interest in this feature, please give this a try and let me know if it works for you

**Comment by cosmin** [comment]
cc @sr1990 since you had some feedback on the earlier PR

**Comment by sr1990** [comment]
> `--use_dovi_supplemental_codecs` (off by default)

Agree with the flag usage. Overall lgtm.

File: packager/mpd/base/mpd_builder.cc [hunk 1]
```diff
@@ -53,12 +53,12 @@ bool AddMpdNameSpaceInfo(XmlNode* mpd) {
       "urn:marlin:mas:1-0:services:schemas:mpd";
   static const char kXmlNamespaceXlink[] = "http://www.w3.org/1999/xlink";
   static const char kMsprNamespace[] = "urn:microsoft:playready";
+  static const char kScte214Namespace[] = "scte214";
```
Comment (vish91) [inline packager/mpd/base/mpd_builder.cc:56]: wonder if this is the correct namespace to use @cosmin  ? 
or if this should be `urn:scte:dash:scte214-extensions`
```
4.2.4 Guidelines for signaling cross-compatible Dolby Vision content (profiles
8.1 and 8.4)
Follow the guidelines for MPD schema and AdaptationSet when signaling cross-compatible Dolby Vision
content.
MPD
An MPD that needs to use this signaling must contain the following schema.
xmlns:scte214="urn:scte:dash:scte214-extensions"
```

**Comment by vish91** [comment]
@XingzhaoYun  could you also help please check this PR and see if you have any feedback or any changes we have in fork where we are attempting this.

**Comment by XingzhaoYun** [comment]
For DASH signaling, I think SupplementalProperty/EssentialProperty for MatrixCoefficients, ColourPrimaries, and TransferCharacteristics are ought to included by default, right? We created another branch including this already for reference, https://github.com/shaka-project/shaka-packager/compare/main...DolbyLaboratories:shaka-packager:dolby/dovi_for_prmt#diff-6f4988849e3a9b9b9060eb05e0dd074599a84197063016b97b15cc71fee287c9

**Comment by cosmin** [comment]
> For DASH signaling, I think SupplementalProperty/EssentialProperty for MatrixCoefficients, ColourPrimaries, and TransferCharacteristics are ought to included by default, right? 

@XingzhaoYun thank you for the feedback, I went ahead and included the changes to parse and include matrix coefficients and color primaries in the MPD.

**Review by joeyparrish — APPROVED** [review]

**Comment by vish91** [comment]
@cosmin  @joeyparrish  how do we use this new flag ? 
I am trying to run a test today and the command when ran with `--use_dovi_supplemental_codecs` says `ERROR: Unknown command line flag 'use_dovi_supplemental_codecs'` 
this is a pacakger command flag right ?

**Comment by cosmin** [comment]
@vish91 did that work out?

**Comment by vish91** [comment]
yeah that worked well. Generated a clear and DRM version. Getting it to player and client teams for testing . Thanks!

**Pull request merged**